packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -447,69 +761,307 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+  _RenderCupertinoDialogActions({
+    List<RenderBox> children,
+    bool isStacked = false,
+  }) : _isStacked = isStacked {
+    addAll(children);
+  }
+
+  bool _isStacked;
+
+  bool get isStacked => _isStacked;
+
+  set isStacked(bool newValue) {
+    if (newValue == _isStacked) {
+      return;
+    }
+
+    _isStacked = newValue;
+    markNeedsLayout();
+  }
 
   @override
-  void paint(Canvas canvas, Size size) {
-    final Paint paint = new Paint()..color = _kButtonDividerColor;
-    canvas.drawLine(const Offset(0.0, 0.0), new Offset(size.width, 0.0), paint);
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! MultiChildLayoutParentData)
+      child.parentData = new MultiChildLayoutParentData();
   }
 
   @override
-  bool shouldRepaint(_CupertinoVerticalDividerPainter other) => false;
-}
+  double computeMinIntrinsicWidth(double height) {
+    return _kCupertinoDialogWidth;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    return _kCupertinoDialogWidth;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    double height = 0.0;
+
+    if (_isStacked) {
+      // Actions min height is first item height + 50% second item height.
+      final RenderBox divider = _findDivider();
+      final double dividerHeight = divider.computeMinIntrinsicHeight(width);
+      final List<RenderBox> firstTwoButtons = _findFirstNButtonRows(2);
+      final double firstButtonHeight = firstTwoButtons[0].computeMinIntrinsicHeight(width);
+      final double secondButtonHeight = firstTwoButtons[1].computeMinIntrinsicHeight(width);
+      height = firstButtonHeight + (0.5 * secondButtonHeight) + (dividerHeight * 2);
+    } else {
+      // Actions min height is first item height (because we are only showing 1 row).
+      final List<RenderBox> firstButton = _findFirstNButtonRows(1);
+      height = firstButton.first.computeMinIntrinsicHeight(width);
+    }
+
+    if (height.isFinite)
+      return height;
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return _findActionsChild().computeMaxIntrinsicHeight(width);
+  }
+
+  @override
+  void performLayout() {
+    // First, layout all children at size zero because if we fail to call
+    // layout() on a child then we'll get an error.
+    final List<RenderBox> children = getChildrenAsList();
+    for (RenderBox child in children) {
+      child.layout(
+        new BoxConstraints.tight(const Size(0.0, 0.0)),
+        parentUsesSize: false,
+      );
+    }
+
+    // Layout the one child we actually care about, which is the dialog "actions".
+    final RenderBox actions = _findActionsChild();
+    actions.layout(
+      constraints,
+      parentUsesSize: true,
+    );
+
+    // Set our size to be as wide as we want, and as tall as the "actions" we
+    // measured.
+    size = new Size(constraints.biggest.width, actions.size.height);
+  }
+
+  RenderBox _findActionsChild() {
```

Comment:
Not clear if you're looking for the child wrapped with the BaseLayoutId or one of the other children.